actor/v7action/service_instance_details.go

```diff
@@ -142,17 +157,19 @@ func (actor Actor) getServiceInstanceDetails(serviceInstanceName string, spaceGU
 	return result, warnings, nil
 }
 
-func (actor Actor) getServiceInstanceParameters(serviceInstanceGUID string) (ServiceInstanceParameters, ccv3.Warnings) {
+func (actor Actor) getServiceInstanceParameters(serviceInstanceGUID string) (ServiceInstanceParameters, ccv3.Warnings, error) {
 	params, warnings, err := actor.CloudControllerClient.GetServiceInstanceParameters(serviceInstanceGUID)
-	if err != nil {
-		if e, ok := err.(ccerror.V3UnexpectedResponseError); ok && len(e.Errors) > 0 {
-			return ServiceInstanceParameters{MissingReason: e.Errors[0].Detail}, warnings
-		} else {
-			return ServiceInstanceParameters{MissingReason: err.Error()}, warnings
-		}
+
+	switch err := err.(type) {
+	case nil:
+	case ccerror.ResourceNotFoundError,
+		ccerror.ServiceInstanceParametersFetchNotSupportedError:
+		return ServiceInstanceParameters{}, warnings, actionerror.ServiceInstanceParamsFetchingNotSupportedError{}
+	default:
+		return ServiceInstanceParameters{}, warnings, err
 	}
 
-	return ServiceInstanceParameters{Value: params}, warnings
+	return ServiceInstanceParameters{Value: params}, warnings, nil
```

Comment:
Given that this is just one line, I'd move it into the switch statement, and then there's no need for a return at the end of the function.